Automattic/wp-calypso
# Hosting Metrics: Prototype first frontend chart

This PR is related to https://github.com/Automattic/dotcom-forge/issues/3051

## Proposed Changes

This PR adds a chart of the data coming from WP.com REST API to `site-monitoring/yousite` page hidden by the feature flag in Calypso.

The chart is populated by the data coming from https://developer.wordpress.com/docs/api/console/ and is using the metric - 'requests_persec' to display data. 

## Testing Instructions

* Make sure you have an Atomic site
* Navigate to Calypso live link and amend `site-monitoring/{yourAtomicsite}`
* View the graph that represents HTTP requests per second on the site in the last 24 hours

<img width="871" alt="Screenshot 2023-07-27 at 9 24 22 PM" src="https://github.com/Automattic/wp-calypso/assets/25575134/6e817816-c172-47de-988c-525202e5ff24">

* If you hover over the data points, it will also highlight the date like indicated on the screenshot. I left it as day/month/year for now there as we don't have any other indication of the exact date on the screen but I would be happy to adjust/remove that:

<img width="883" alt="Screenshot 2023-07-27 at 9 26 42 PM" src="https://github.com/Automattic/wp-calypso/assets/25575134/74b6181c-1b92-4ad1-bf34-69e628574670">

**Extra notes**:

* I used `useMemo` for the `startTime` and `endTime` to prevent the graph from updating in the real time and only do it when the page is re-loaded (since the start and end times are constantly changing, the query keeps on being constantly sent to the API). I tried using `useEffect` instead to prevent the graph from constantly re-rendering but it makes writing tests quite tricky. There is probably a better strategy for this so happy to implement any other suggestions!
* This still needs tests - it is WIP.

## Pre-merge Checklist
